### PR TITLE
fix(orders-table): display fulfilled status before cancelling

### LINKED ISSUE #2710: Order's part hangs in 'Cancelling' state when filled
Related to #2633 

1. Place a TWAP order with more than 2 parts
2. Wait till the 1st part is filled
3. Then cancel the Main (parent) order

**AR**: the filled part changes the state to 'Cancelling' in the History (and hangs with this status)

**ER**: the filled part is Filled.

### PR DESCRIPTION
**Pull request by shoom3301**
# Summary

Fixes #2710, #2802

## Changes
1. `ComposableCowInfo.isVirtualPart` to clearly distinguish part orders from discrete orders
2. Added `scheduled` section to orders state
3. Added moo sound and popup for TWAP order placement (`dispatchPresignedOrderPosted`)
4. Change all child orders statuses to Canceled when the parent is canceled (`cancelOrderInState`)
5. Change all child orders statuses to Cancelling when the parent is canceling (`_transformGpOrderToStoreOrder`)
6. Display warnings in the orders table only for Pending orders
7. Change TWAP order start date after tx mining from tx submitted date to tx executed date (`emulateTwapAsOrder`)
8. Remove rejected (in Safe) orders (`ordersToDelete`)

Some code here is not the best, sorry for that. I will be happy to refactor it after the Beta release

### CODE REVIEW

**Review by elena-zh — CHANGES_REQUESTED** [review]
Hey @shoom3301 , something is wrong with orders statuses: as soon as order is signed, all orders  (including the parent one) go to 'Scheduled' status. 
When a part is filled, it is not reflected on the UI until I refresh the page. 
Once refresh, I can see the parent order blinking with 'Open' status, then returning back to 'Scheduled'. 
See the video: https://www.loom.com/share/a7005a46ceac41e1b97c7707c2be1c11

Somehow I caught up the possibility yo cancel the parent order (quickly pressed on the Cancel button for the blinking order), and noticed that the filled part did not change its status. However, the parent order did not show the 'filled' info (33% filled): 
image

**Review by alfetopito — APPROVED** [review]

**Comment by shoom3301** [comment]
@elena-zh it seems to be unrelated to #2710 issue, doesn't it?

**Comment by elena-zh** [comment]
> @elena-zh it seems to be unrelated to #2710 issue, doesn't it?

Not sure. How can I properly test statuses as I can hardly catch 'Cancel order' button due to all orders are in 'Scheduled' state?

**Comment by shoom3301** [comment]
@elena-zh let's do a call, I can't reproduce it

**Comment by elena-zh** [comment]
Hey @shoom3301 , 
I cleared my local storage and did another round of testing. 
Good news, the 'Filled' order status is not changed when I cancel a parent order!

However. there are some other issues with 'Cancel' feature that I faced: 
1. When a part is in Open state, and I cancel it, once TX is mined the order goes to 'Cancelled' state even if it is filled. Can we leave 'Filled' status for the part even if Cancel TX is successfully executed for the part?
ffilled
2. When a parent order is in Cancelling state, its part is still in Scheduled
not cancelling
3. When a scheduled part is cancelled, it navigates to 404 page on Explorer
cancelled1

LMK please if I need to open a separate issue on any of these items.

**Comment by shoom3301** [comment]
More issues:
https://github.com/cowprotocol/cowswap/pull/2764#issuecomment-1622082461

**Comment by shoom3301** [comment]
@elena-zh fixed

**Comment by elena-zh** [comment]
Hey @shoom3301 , issues that I faced: 
1. When I cancel the past order's part, I see that the parent order hangs in the 'PF' status in the Open orders tab. It should go to the 'History' tab
cancelled when the last part

2. I don't see any pop-ups and sounds here (when an order is placed/filled). Pop-ups appear only when cancel order
3. Time of the parent part is a bit different in comparison with the last its part
not the same
4. I created #2802 issue as an edge case when execute a TWAP order TX when it is expired
5. I still can see order's part appearing in the Open order tabs after cancelling the main order 
cancelled

File: src/legacy/state/orders/reducer.ts [hunk 1]
```diff
@@ -387,12 +403,31 @@ export default createReducer(initialState, (builder) =>
...
+              ...ordersMap.expired,
+              ...ordersMap.cancelled,
+              ...ordersMap.creating,
+              ...ordersMap.failed,
+              ...ordersMap.scheduled,
+            }
 
-          orderObject.order.status = OrderStatus.CANCELLED
-          orderObject.order.isCancelling = false
+            const children = Object.values(allOrdersMap).filter(
```
Comment (nenadV91) [inline src/legacy/state/orders/reducer.ts:421]: Could you do this in one go, for example:

```javascript
Object.values(allOrdersMap).forEach((item) => {
  if (item?.order.composableCowInfo?.parentId === id) {
    cancelOrderInState(state, chainId, item)
  }
})
```

**Review by nenadV91 — APPROVED** [review]
LGTM 👍

**Review by elena-zh — APPROVED** [review]
Positive flow works fine!
There are still some issues related to edge cases: 
 - #2808 
 - #2809 
 - #2715 
 -  #2792

**Pull request merged**